setup.py

```diff
@@ -7,6 +7,7 @@
     "thrift~=0.13",
     "boto3~=1.9",
     "pyee~=6.0",
+    "psycopg~=3.0.10",
```

Comment:
I am a little worried about dependency version conflicts here. It's also a bit of a weighty dependency for someone who doesn't want to use it.

The easy workaround is just to add a new `extras_require` section for "postgres, i.e. `pip install flipper-client[postgres]`. That allows users to opt-in if they want to use it. Honestly, we should probably do this for all the backends we support.

Reply:
Great that you touched this topic. I meant to get some feedback on this: 
`psycopg` requires `libpq` to be installed in the system. Because we import every backend to the namespace, a user who isn't using the postgreSQL backend will still need to have `libpq` installed otherwise the import of flipper fails.

I couldn't think of a way to fix this that doesn't break backwards compatibility

Reply:
Yeah I think the easiest thing to do is to add a separate `postgres` target to the `extras_require` section. That forces the user to opt-in to installing postgres. Once opted-in, then we can add instructions to the readme around specific installation requirements for postgres.

Reply:
Oh, got it. Thanks for the clarification. I can think of two ways around this. Python (for better or worse) allows you to import dependencies inline (In a method, for example). You can always try/except too. I think this is my preferred approach:

```python
try:
  import psycopg
except ModuleNotFoundError:
  pass
```

Reply:
If you want to be extra careful, you could even set a module-level flag:

```python
POSTGRES_ENABLED = False
try:
  import psycopg
  POSTGRES_ENABLED = True
except ModuleNotFoundError:
  pass


class PostgreSQLFeatureFlagStore:
  # ...
  def __init__(self, *args, **kwargs):
    if POSTGRES_ENABLED is False:
       raise PostgresNotEnabled()
```

Reply:
I used your approach but in the `contrib/__init__.py` file (I think its more extendable for the future with other optional stores), let me know what you think.

I really appreciate the time your taking to give feedback :bow: